assets/js/linear_algebra/dot_product_collide2d.js

```diff
@@ -0,0 +1,336 @@
+let dot_product_collide2d = (function() {
+
+let origin = [300, 140], 
+    scale = 60, 
+    scatter = [], 
+    axis = [],
+    expectedAxis = [],
+    startAngleX = Math.PI,
+    startAngleY = 0.,
+    startAngleZ = 0.,
+    axis_len = 2,
+    unit = axis_len/10,
+    svg = null,
+    lib = null,
+    uTvv_opacity = 0;
+
+
+function select_svg(svg_id){
+  svg = d3.select(svg_id);
+  lib = space_plot_lib(
+      svg,
+      origin, 
+      scale,
+      is_2d=true);
+
+  svg = svg.call(d3.drag()
+         .on('drag', dragged)
+         .on('start', drag_start)
+         .on('end', drag_end))
+         .append('g');
+}
+
+
+function plot(scatter, axis, tt){
+  let u = scatter[0],
+      v = scatter[1];
+
+  basis = {
+      ex: axis[1/unit -1 + axis_len/unit * 0][1], 
+      ey: axis[1/unit -1 + axis_len/unit * 1][1],
+      ez: {x: 0, y: 0, z: 0}, // dummy 
```

Comment:
please put the correct formula for ez here instead of the dummy.

Reply:
and in other *2d.js scripts as well.

Reply:
done